lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -707,9 +734,29 @@ impl LayerModel {
         }
     }
 
-    /// Create a new sublayer to this layer, with the same camera.
+    /// Create a new sublayer to this layer. It will inherit this layer's camera.
     pub fn create_sublayer(&self, name: impl Into<String>) -> Layer {
-        let layer = Layer::new_with_cam(name, &self.camera.borrow());
+        let layer = Layer::new_with_flags(name, true, true);
+        self.add_sublayer(&layer);
+        layer
+    }
+
+    /// Create a new sublayer to this layer. Override the camera for this layer, breaking the camera
+    /// inheritance chain. Updates to the parent camera will not affect this layer.
+    pub fn create_sublayer_with_cam(&self, name: impl Into<String>, camera: &Camera2d) -> Layer {
+        let layer = Layer::new_with_cam(name, camera);
+        self.add_sublayer(&layer);
+        layer
+    }
+
+
+    /// Create a new sublayer to this layer. It will inherit this layer's camera, but will not be
+    /// rendered as a part of standard layer stack. Instead, it will only be rendered as a mask for
+    /// other layers. See [`Layer::set_mask`] for more information. Note that this will not set up
+    /// the mask connection. You still have to separately call `set_mask` on any layer you want to
+    /// mask with this layer.
+    pub fn create_mask_sublayer(&self, name: impl Into<String>) -> Layer {
+        let layer = Layer::new_with_flags(name, true, false);
```

Comment:
I do not understand this - mask layers are also renderable, because the mask has to be rendered to an offscreen texture fotit to be applied,so why masked layers have renderable flag set to false?

Reply:
I think the name of that flag is too ambiguous, as you already pointed out. Maybe `render_only_as_mask` would be better.

Reply:
Changed that to `MAIN_PASS_VISIBLE` bitflag and added docs to explain it further.